server/src/main/java/io/whitefox/api/deltasharing/loader/DeltaSharedTable.java

```diff
@@ -0,0 +1,60 @@
+package io.whitefox.api.deltasharing.loader;
+
+import io.delta.standalone.DeltaLog;
+import io.delta.standalone.Snapshot;
+import io.delta.standalone.actions.Metadata;
+import io.whitefox.persistence.memory.PTable;
+import java.nio.file.Path;
+import java.sql.Timestamp;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+import org.apache.hadoop.conf.Configuration;
+
+public class DeltaSharedTable {
+
+  private final PTable pTable;
+  private final DeltaLog deltaLog;
+  private final Configuration configuration;
+  private final Path dataPath;
+
+  public DeltaSharedTable(PTable pTable) {
+    this.pTable = pTable;
+    this.configuration = new Configuration();
+    this.dataPath = Path.of(pTable.location());
+    this.deltaLog = deltaLog();
+  }
+
+  public DeltaLog deltaLog() {
```

Comment:
Instead you could return a CompletionStage so that it handles exceptions “by design”